### PR TITLE
AV1: Added OBU parsing and frame data parsing.

### PR DESCRIPTION
**Pull request by jeffqjiangNew**

### CODE REVIEW

File: src/parser/av1_parser.cpp [hunk 1]
```diff
@@ -39,11 +39,150 @@ rocDecStatus Av1VideoParser::UnInitialize() {
...
 
 rocDecStatus Av1VideoParser::ParseVideoData(RocdecSourceDataPacket *p_data) { 
-    //to be implemented
-    return ROCDEC_NOT_IMPLEMENTED;
+    if (p_data->payload && p_data->payload_size) {
+        if (ParsePictureData(p_data->payload, p_data->payload_size) != PARSER_OK) {
+            ERR(STR("Parser failed!"));
+            return ROCDEC_RUNTIME_ERROR;
+        }
+
```
Comment (rrawther) [inline src/parser/av1_parser.cpp]: remove empty line

File: src/parser/av1_parser.cpp [hunk 2]
```diff
@@ -39,11 +39,150 @@ rocDecStatus Av1VideoParser::UnInitialize() {
...
+ParserResult Av1VideoParser::ParsePictureData(const uint8_t *p_stream, uint32_t pic_data_size) {
+    ParserResult ret = PARSER_OK;
+
+    pic_data_buffer_ptr_ = (uint8_t*)p_stream;
+    pic_data_size_ = pic_data_size;
+    curr_byte_offset_ = 0;
+
+    do {
+        if ((ret = ReadObuHeaderAndSize()) == PARSER_EOF ) {
```
Comment (rrawther) [inline src/parser/av1_parser.cpp]: probably you can change it to while (cond){ } instead and add is as part of cond.

File: src/parser/av1_parser.cpp [hunk 3]
```diff
@@ -39,11 +39,150 @@ rocDecStatus Av1VideoParser::UnInitialize() {
...
+    obu_header_.obu_has_size_field = Parser::GetBit(p_stream, offset);
+    if (!obu_header_.obu_has_size_field) {
+        ERR("Syntax error: Section 5.2: obu_has_size_field must be equal to 1.");
+        return PARSER_INVALID_ARG;
+    }
+    if (Parser::GetBit(p_stream, offset) != 0) {
+        ERR("Syntax error: obu_reserved_1bit must be set to 0.");
+        return PARSER_INVALID_ARG;
+    }
+
```
Comment (rrawther) [inline src/parser/av1_parser.cpp]: do we need empty line here?

**Review by rrawther — CHANGES_REQUESTED** [review]
Added some comments.
I see the use of lots of empty lines in the code, @AryanSalmanpour: please check if this is up  to standard convention. Or else, remove un-necessary empty lines

File: src/parser/av1_parser.cpp (hunk 2 above)
Comment (jeffqjiangNew) [reply src/parser/av1_parser.cpp]: Yes. Was preparing something more complicated but now it is not needed. Done.

File: src/parser/av1_parser.cpp (hunk 1 above)
Comment (jeffqjiangNew) [reply src/parser/av1_parser.cpp]: Done.

File: src/parser/av1_parser.cpp (hunk 3 above)
Comment (jeffqjiangNew) [reply src/parser/av1_parser.cpp]: Removed.

**Comment by jeffqjiangNew** [comment]
@rrawther Please take a look the latest changes.

File: src/parser/av1_parser.cpp [hunk 4]
```diff
@@ -194,7 +315,27 @@ void Av1VideoParser::ParseSequenceHeader(uint8_t *p_stream, size_t size) {
     p_seq_header->film_grain_params_present = Parser::GetBit(p_stream, offset);
 }
 
-ParserResult Av1VideoParser::ParseUncompressedHeader(uint8_t *p_stream, size_t size) {
+ParserResult Av1VideoParser::ParseFrameHeaderObu(uint8_t *p_stream, size_t size, int *p_bytes_parsed) {
+    if (seen_frame_header_ == 1) {
```
Comment (AryanSalmanpour) [inline src/parser/av1_parser.cpp:319]: Are you planning to add any code to the body of this if condition `if (seen_frame_header_ == 1) {` ? if not, 
Please change this if-else condition to an if condition only -> `if (frame_header_ > 1) {`

Comment (jeffqjiangNew) [reply src/parser/av1_parser.cpp:319]: At this moment, not sure if we need to do something here (actually copy the frame header or not). Will check during bring up.

**Review by rrawther — APPROVED** [review]

**Review by AryanSalmanpour — APPROVED** [review]

**Pull request merged**